### PR TITLE
refactor: enhance cli

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
1. cli wait for the tls handshake timeout while the remote server is unreachable
2. show version compare log while listing resource

**Solution:**
1. reduce the timeout by adding dial to check reachable, reduce the tls handshake timeout and remove duplicate validation multiple times
2. remove version compare while listing resource
3. remove env clone command

**Related Issue:**
#1960 
#1959

### CODE REVIEW

File: pkg/cli/cmd/version.go [hunk 1]
```diff
@@ -29,6 +29,11 @@ func Version(sc *config.Config) *cobra.Command {
 			}
 
 			fmt.Print(string(b))
+
+			err = api.CompareVersion(info)
+			if err != nil {
+				panic(err)
+			}
```
Comment (gitlawr) [inline pkg/cli/cmd/version.go:36]: Can we skip this check? Just print the version info.

File: pkg/cli/config/config.go [hunk 2]
```diff
@@ -71,18 +71,46 @@ func (c *Config) DoRequest(req *http.Request) (*http.Response, error) {
...
+
 	msg := `cli configuration is invalid: %s. You can configure cli by running "walrus login"`
 
-	if c.Server == "" {
-		return fmt.Errorf(msg, "server address is empty")
+	switch {
+	case c.Server == "":
+		err = fmt.Errorf(msg, "server address is empty")
+	case c.Token == "":
+		err = fmt.Errorf(msg, "token is empty")
```
Comment (gitlawr) [inline pkg/cli/config/config.go:90]: In local docker mode, authn is disabled and empty token is used

File: pkg/cli/config/config.go [hunk 3]
```diff
@@ -70,20 +70,57 @@ func (c *Config) DoRequest(req *http.Request) (*http.Response, error) {
...
 	}
 
-	if c.Token == "" {
-		return fmt.Errorf(msg, "token is empty")
+	err = c.CheckReachable()
```
Comment (alexcodelf) [inline pkg/cli/config/config.go:119]: Is it intend to overwirte err?

File: pkg/cli/config/config.go [hunk 4]
```diff
@@ -115,7 +153,7 @@ func (c *Config) HttpClient(timeout time.Duration) *http.Client {
 		DialContext: (&net.Dialer{
 			Timeout: 3 * time.Second,
 		}).DialContext,
-		TLSHandshakeTimeout: 10 * time.Second,
+		TLSHandshakeTimeout: 5 * time.Second,
```
Comment (thxCode) [inline pkg/cli/config/config.go:156]: why do we need to change this?

File: pkg/cli/config/config.go [hunk 5]
```diff
@@ -94,7 +131,8 @@ func (c *Config) ValidateAndSetup() error {
 	}
 
 	if err != nil {
-		return fmt.Errorf(msg, err.Error())
+		err = fmt.Errorf(msg, err.Error())
+		return
```
Comment (thxCode) [inline pkg/cli/config/config.go:135]:
```
return fmt.Errorf(`cli configuration is invalid: %w. You can configure cli by running "walrus login"`, err)
```
or 
```
return fmt.Errorf(`cli configuration is invalid: %v. You can configure cli by running "walrus login"`, err)
```

File: pkg/cli/config/server_context.go [hunk 6]
```diff
@@ -26,6 +26,9 @@ type ServerContext struct {
 	Server   string `json:"server" survey:"server"`
 	Token    string `json:"token" survey:"token"`
 	Insecure bool   `json:"insecure" survey:"insecure"`
+
+	// Reachable.
+	Reachable bool
```
Comment (thxCode) [inline pkg/cli/config/server_context.go:31]: can we use `Unreachable bool` here?

File: pkg/cli/config/config.go (hunk 3 above)
Comment (aiwantaozi) [reply pkg/cli/config/config.go:119]: updated

File: pkg/cli/config/server_context.go (hunk 6 above)
Comment (aiwantaozi) [reply pkg/cli/config/server_context.go:31]: Prefer positive and affirmative variable names, Reachable might be more intuitive.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**